core/src/env_platform/db_handler.bal

```diff
@@ -543,3 +544,165 @@ function isMinistryHasUser(string ministryId, string userId) returns boolean|err
+function isApplicationRelatedTo(string userType, string userId, string applicationId) returns boolean|error {
+    if (userType == "Admin") {
+        return true;
+    } else if (userType == "User") {
+        return applicationBelongsToUser(applicationId, userId);
+    } else if (userType == "Ministry") {
+        string ministryRelatedToUser = check getMinistryRelatedToUser(userId);
+        return isMinistryAssigned(applicationId, ministryRelatedToUser);
+    } else {
+        return false;
+    }
+}
+
+# The `getMinistryRelatedToUser` function will output the ministry which the given user is registered to.
+# 
+# + userId - ID of the user.
+# + return - This function will return either the ministry ID or an appropriate error.
+function getMinistryRelatedToUser(string userId) returns string|error {
+    map<json>[] ministries = check ministryCollection->find();
+    foreach map<json> ministry in ministries {
+        json[]|error users = trap <json[]>ministry.users;
+        // If ministry does not contains users.
+        if (users is error) {
+            continue;
+        } else {
+            foreach json user in users {
+                if (user.id == userId) {
+                    log:printDebug("User with user ID " + userId + " is a member of ministry with ID "
+                        + ministry.id.toString() + ".");
+                    return ministry.id.toString();
+                }
+            }
+        }
+    }
+    return error("Not found", message = "Ministry is not found related to the user with ID " + userId + ".");
+}
+
+# The `isMinistryAssigned` function will check whether the given ministry is assigned to the given application.
+# 
+# + applicationId - ID of the application.
+# + ministryId - ID of the ministry.
+# + return - This function will return whether the ministry is assigned or an appropriate error.
+function isMinistryAssigned(string applicationId, string ministryId) returns boolean|error {
+    map<json>[] applications = check applicationCollection->find({"applicationId": applicationId, status: "submit"});
+    if (applications.length() == 0) {
+        return error("Not found", message = "Application is not found");
+    } else {
+        map<json> application = <map<json>>applications[0];
+        json[]|error assignments = trap <json[]>application.assignments;
+        if (assignments is error) {
+            return false;
+        } else {
+            foreach json assignment in assignments {
+                if (assignment.id == ministryId) {
+                    log:printDebug("Ministry with ID " + ministryId + "is assiged to the application with ID "
+                        + assignment.id.toString() + ".");
+                    return true;
+                }
+            }
+        }
+    }
+    return false;
+}
+
+# The `getComment` function will return the comment with the ID for the given application with application ID.
```

Comment:
@yashodgayashan try to make the method description simple to understand. We don't need to say everything about it. For example, we don't need to point to application ID put it in a general way. Check this in other places as well.
```suggestion
# The `getComment` function returns the comment which is related to the application based on the provided arguments.
```